src/components/smart_objects/src/array_schema_item.cc

```diff
@@ -90,6 +92,31 @@ errors::eType CArraySchemaItem::validate(
   return errors::OK;
 }
 
+bool CArraySchemaItem::filterInvalidEnums(
+    SmartObject& Object,
+    const utils::SemanticVersion& MessageVersion,
+    rpc::ValidationReport* report__) {
+  if (SmartType_Array == Object.getType()) {
+    int index = 0;
+    auto array = Object.asArray();
+    for (auto it = array->begin(); it != array->end();) {
+      std::stringstream strVal;
+      strVal << index;
+      if (mElementSchemaItem->filterInvalidEnums(
+              *it, MessageVersion, &report__->ReportSubobject(strVal.str()))) {
+        it = array->erase(it);
+      } else {
+        ++it;
+      }
+      index++;
+    }
+    if (array->empty()) {
```

Comment:
Should this method return true if there was at least enum item removed?

Reply:
No, that would mean that the array should be removed, which doesn't make sense in that case

Reply:
That said, it would probably be better to return if the list is shorter than the minimum length, rather than explicitly checking for an empty list

Reply:
Ah, I think I see your point, it should return true ONLY if at least one item is removed. Fixed in https://github.com/smartdevicelink/sdl_core/pull/3385/commits/bee65b5d8fe27964ed09db991e9ecd8c3b7d8d2c